homeassistant/components/stream/__init__.py

```diff
@@ -87,13 +87,14 @@ def create_stream(
 
     # For RTSP streams, prefer TCP
     if isinstance(stream_source, str) and stream_source[:7] == "rtsp://":
-        options = {
+        stream_options = {
             "rtsp_flags": "prefer_tcp",
             "stimeout": "5000000",
-            **options,
         }
 
-    stream = Stream(hass, stream_source, options=options, stream_label=stream_label)
+    stream = Stream(
+        hass, stream_source, options=stream_options, stream_label=stream_label
```

Comment:
So the idea would be in the future that only supported options would be passed through. (e.g. we device to support option X and make it use a more friendly name that makes sense in context, or could map to many PyAV options if that made sense). Am I understanding that right? If so, makes a lot of sense to me.

What happens to the existing uses of stream_options that get passed into the `options` parameter? https://github.com/home-assistant/core/search?q=stream_options  I'm missing something.

Reply:
Sorry you're right, generic and onvif currently use it (through Camera). So we'd have to modify those uses too. But yes, the general idea is what you wrote above.

Reply:
Great, that sounds like a good direction! Just spit balling here, one suggestion could be to just have a filtered list of options or make another object (e.g. a data class or something) of options that is translated into the pyav options.